addons/machinecontroller/machine-controller.yaml

```diff
@@ -0,0 +1,672 @@
+      - "certificatesigningrequests/approval"
+    verbs:
+      - "update"
+  - apiGroups:
+      - "certificates.k8s.io"
+    resources:
+      - "signers"
+    resourceNames:
+      - "kubernetes.io/kubelet-serving"
+    verbs:
+      - "approve"
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: machine-controller
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: machine-controller
+subjects:
+  - kind: ServiceAccount
+    name: machine-controller
+    namespace: kube-system
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: machine-controller-webhook
+  namespace: kube-system
+spec:
+  ports:
+    - port: 443
+      protocol: TCP
+      targetPort: 9876
+  selector:
+    app: machine-controller-webhook
+  type: ClusterIP
+
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: machine-controller
+  namespace: kube-system
+
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: machine-controller
+  namespace: kube-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: machine-controller
+  template:
+    metadata:
+      annotations:
+        "prometheus.io/scrape": "true"
+        "prometheus.io/port": "8080"
+        "prometheus.io/path": "/metrics"
+        "caBundle-hash": "{{ .Config.CABundle | sha256sum }}"
+      labels:
+        app: machine-controller
+    spec:
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      tolerations:
+        - key: "node-role.kubernetes.io/master"
+          operator: Exists
+          effect: NoSchedule
+        - key: "node.cloudprovider.kubernetes.io/uninitialized"
+          value: "true"
+          effect: NoSchedule
+        - key: "CriticalAddonsOnly"
+          operator: Exists
+      serviceAccountName: machine-controller
+      containers:
+        - name: machine-controller
+          image: "{{ .InternalImages.Get "MachineController" }}"
+          imagePullPolicy: IfNotPresent
+          command:
+            - /usr/local/bin/machine-controller
+            - -logtostderr
+            - -v=4
+            - -health-probe-address=0.0.0.0:8085
+            - -metrics-address=0.0.0.0:8080
+            - -cluster-dns={{ .Resources.NodeLocalDNSVirtualIP }}
+            - -node-csr-approver
+            - -join-cluster-timeout=15m
+            - -node-container-runtime={{ .Config.ContainerRuntime }}
+            {{ with .Config.Proxy.HTTP }}
+            - -node-http-proxy={{ . }}
+            {{ end }}
+            {{ with .Config.Proxy.NoProxy }}
+            - -node-no-proxy={{ . }}
+            {{ end }}
+            {{ with .Config.RegistryConfiguration.InsecureRegistryAddress }}
+            - -node-insecure-registries={{ . }}
+            {{ end }}
+            {{ if .Config.CABundle }}
+            - -ca-bundle={{ .Resources.CABundleSSLCertFilePath }}
+            {{ end }}
+            # TODO: handle dynamic pause-image
+            # - -node-pause-image=
```

Comment:
Can we create a follow-up issue for this?

Reply:
I've fixed it